src/api/specials/diary/controller.js

```diff
@@ -1,59 +1,119 @@
   try {
     const today = buildColloquialDateHash(month, day);
-    const diary = await Diary.findOne({ locale, colloquialDateHash: today, active: true }).exec();
-
-    let [bestPrev] = await Diary.find({ locale, colloquialDateHash: { $lt: today } })
-      .sort({
-        colloquialDateHash: 'desc',
-      })
-      .limit(1)
-      .exec();
+    const diaries = await Diary.find({ colloquialDateHash: today, active: true }).populate(
+      'author'
+    );
+    const diary = sample(diaries);
+
+    let [bestPrev] = await Diary.find({ colloquialDateHash: { $lt: today } })
+      .sort({ colloquialDateHash: 'desc' })
+      .limit(1);
+
     if (!bestPrev) {
       // The very beginning of the year might be requested.
-      [bestPrev] = await Diary.find({ locale })
+      [bestPrev] = await Diary.find({})
         .sort({ colloquialDateHash: 'desc' })
-        .limit(1)
-        .exec();
+        .limit(1);
+
       if (!bestPrev) {
         // We have no diaries at all.
         throw new HttpError(HttpStatus.NO_CONTENT, 'errors.diariesMissing');
       }
     }
 
-    let [bestNext] = await Diary.find({ locale, colloquialDateHash: { $gt: today } })
+    let [bestNext] = await Diary.find({ colloquialDateHash: { $gt: today } })
       .sort({ colloquialDateHash: 'asc' })
-      .limit(1)
-      .exec();
+      .limit(1);
+
     if (!bestNext) {
       // The very end of the year might be requested.
-      [bestNext] = await Diary.find({ locale })
+      [bestNext] = await Diary.find({})
         .sort({ colloquialDateHash: 'asc' })
-        .limit(1)
-        .exec();
+        .limit(1);
     }
 
     return sendJson(res)({
       data: serializeDiary(diary),
-      prev: !!bestPrev && {
-        month: bestPrev.month,
-        day: bestPrev.day,
-      },
-      next: !!bestNext && {
-        month: bestNext.month,
-        day: bestNext.day,
-      },
+      prev: getQuery(bestPrev),
+      next: getQuery(bestNext),
     });
   } catch (err) {
     return next(err);
   }
 };
 
-export default getDay;
+export const fiberyImport = async ({ user }, res, next) => {
+  try {
+    // FIXME: metadata.createdAt everytime new
+    const metadata = getInitObjectMetadata(user);
+
+    const data = await fibery.getDiaries();
+    const tagsById = data.reduce((acc, { personality: tag }) => {
+      if (tag) {
+        acc[tag.fiberyId] = tag;
+      }
+      return acc;
+    }, {});
+
+    const tags = Object.values(tagsById).map(getMapTag('personalities'));
+    const authors = await updateTags(tags, metadata, { skipValidation: true, skipMap: true });
+    const authorsIds = authors.reduce(
+      (acc, { fiberyId, _id }) => ({ ...acc, [fiberyId]: _id }),
+      {}
+    );
+    const diariesData = data.map(({ personality, day, month: { rank }, ...rest }) => ({
+      ...rest,
+      author: authorsIds[personality.fiberyId],
+      // TODO: check if rank is right
+      colloquialDateHash: (rank / ENUM_BASE + 1) * 100 + day,
```

Comment:
is the comment explaining what's `colloquialDateHash` in model file still valid?
the `(month * 100 + day)` one